File: pg-index-health-spring-boot-demo/src/main/java/io/github/mfvanek/pg/index/health/demo/dto/ForeignKeyMigrationResponse.java
/*
 * Copyright (c) 2019-2024. Ivan Vakhrushev and others.
 * https://github.com/mfvanek/pg-index-health-demo
 *
 * Licensed under the Apache License 2.0
 */

package io.github.mfvanek.pg.index.health.demo.dto;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
import io.github.mfvanek.pg.model.constraint.ForeignKey;

import java.util.List;

@SuppressFBWarnings("EI_EXPOSE_REP2")
public record ForeignKeyMigrationResponse(
    List<ForeignKey> foreignKeysBefore,
    List<ForeignKey> foreignKeysAfter,
    List<String> generatedMigrations
) {

}

File: pg-index-health-spring-boot-kotlin-demo/src/main/kotlin/io/github/mfvanek/pg/index/health/demo/kotlin/dto/ForeignKeyMigrationResponse.kt
/*
 * Copyright (c) 2019-2025. Ivan Vakhrushev and others.
 * https://github.com/mfvanek/pg-index-health-demo
 *
 * Licensed under the Apache License 2.0
 */

package io.github.mfvanek.pg.index.health.demo.kotlin.dto

import io.github.mfvanek.pg.model.constraint.ForeignKey
import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Response containing foreign key migration information")
class ForeignKeyMigrationResponse(
    @field:Schema(description = "List of foreign keys before migration")
    val foreignKeysBefore: List<ForeignKey>,
    @field:Schema(description = "List of foreign keys after migration")
    val foreignKeysAfter: List<ForeignKey>,
    @field:Schema(
        description = "List of generated migration scripts",
        example = "[\"ALTER TABLE orders ADD CONSTRAINT fk_orders_buyer_id FOREIGN KEY (buyer_id) REFERENCES buyer(id);\"]"
    )
    val generatedMigrations: List<String>
)
